pkg/clients/iam/iamrole.go

```diff
@@ -198,25 +198,23 @@ func IsRoleUpToDate(in v1beta1.IAMRoleParameters, observed iamtypes.Role) (bool,
 }
 
 // DiffIAMTags returns the lists of tags that need to be removed and added according
-// to current and desired states.
-func DiffIAMTags(local []v1beta1.Tag, remote []iamtypes.Tag) (add []iamtypes.Tag, remove []string) {
-	addMap := make(map[string]string, len(local))
-	for _, t := range local {
-		addMap[t.Key] = t.Value
-	}
+// to current and desired states, also returns if desired state needs to be updated
+func DiffIAMTags(local map[string]string, remote []iamtypes.Tag) (add []iamtypes.Tag, remove []string, areTagsUpToDate bool) {
```

Comment:
Sorry I missed this. This was a pure function, and now mutates the input. I don't know if crossplane has any guidelines around this but I think it is less error prone. Have you checked all callers?

cc @haarchri what do you think?

Reply:
Hey @chlunde yeah I have checked all the callers. I modified this to actually reuse this piece. `IamRoles/IamPolicies` both are using this for diffing the tags.

@haarchri Sir, any inputs ?

Reply:
@chlunde is this a Blocker for you for me test was successful and tests incode looking AS expected

Reply:
@haarchri @ra-grover no, it's good. I see the reason is that local is not a map, it used to be a slice. All callers take a copy of the input before calling this function so there should not be any bugs.

I hope we can make a generic version of DiffTags when Go 1.18 is out.. :)